lib/bolt/shell/powershell.rb

```diff
@@ -14,6 +14,22 @@ def initialize(target, conn)
         extensions = [target.options['extensions'] || []].flatten.map { |ext| ext[0] == '.' ? ext : '.' + ext }
         extensions += target.options['interpreters'].keys if target.options['interpreters']
         @extensions = DEFAULT_EXTENSIONS + extensions
+        validate_ps_version
+      end
+
+      def validate_ps_version
+        version = execute("$PSVersionTable.PSVersion.Major").stdout.string.chomp
+        if !version.empty? && version.to_i < 3
+          # This lets us know how many targets have Powershell 2, and lets the
+          # user know how many targets they have with PS2
+          msg = "Detected Powershell 2 on one or more targets.\nPowershell 2 "\
+            "is deprecated, and support will be removed in Bolt 3.0. See "\
+            "bolt-debug.log or run with '--log-level debug' to see the full "\
+            "list of targets with Powershell 2."
+
+          Bolt::Logger.deprecation_warning("Powershell 2", msg)
+          @logger.debug("Detected Powershell 2 on #{target}.")
```

Comment:
```suggestion
          @logger.debug("Detected PowerShell 2 on #{target}. Support for PowerShell is deprecated.")
```

The debug message on its own doesn't say much currently.

Reply:
Oh it's not meant to. Users will see a warning that Powershell 2 is deprecated, but they'll only be warned once if any targets have Powershell 2. The intention is that users will see the warning and look in debug logs for which targets have Powershell 2. They should already know it's deprecated if they're seeing these logs. I wanted to keep the message short to avoid overloading debug output.

Reply:
I could maybe have it say "Detected unsupported Powershell version 2 on #{target}."

Reply:
I think that would be helpful. Even though there is the warning that gives more context, I think each individual debug message should speak for itself.

Reply:
Also, the check is for < 3, which means PowerShell 1 will also probably trigger this warning. I'm not sure if it's too much of a concern that the warnings say PowerShell 2 though, since 1 is fairly old.

Reply:
*Actually* Powershell one doesn't have `PSVersionTable`, and this won't raise a warning if nothing is returned. Since we don't support Powershell 1 and none of the systems from analytics have Powershell 1 I don't think we should add special handling for it.